doc/source/area-detector.rst

```diff
@@ -269,6 +269,168 @@ The vendor specific details are embedded in the cams
+
+    Areadetector_device_PV_suffix:Stats
+
+And for the builtin areadetector 'color mode' attribute it looks like:
+
+.. code-block:: python
+
+    Areadetector_device_PV_suffix:ColorMode_RBV
+
+where ``Areadetector_device_PV_suffix`` is the PV name for the Area detector
+device, ``plugin_suffix = Stats`` is the 'stats' Plugin suffix and
+``attribute_suffix = ColorMode_RBV`` is the'color mode' attribute suffix.
+
+In order to create the class then the following code is required (where ``XXX``
+is the name of the device):
+
+.. code-block:: python
+
+    from ophyd.areadetector.base import ADComponent as ADCpt, ad_group,
+                                        EpicsSignalWithRBV
+    from ophyd.signal import EpicsSignal, EpicsSignalRO
+    from ophyd.device import DynamicDeviceComponent as DDCpt
+    from ophyd.detectors import DetectorBase
+
+    Class XXX('optional trigger mixin, 'DetectorBase):
+        '''An areadetector device class for ......'''
+        _suffix_re = '"Areadetector_suffix"\d:'
+
+        # ADD ATTRIBUTES AS COMPONENTS HERE USING THE SYNTAX
+        attribute_name = Cpt(Type, attribute_suffix)
+            # where 'Type' is EpicsSignal, EpicsSignalRO, EpicsSignalWithRBV,..
+
+        # ADD ATTRIBUTE GROUPS AS COMPONENTS USING THE SYNTAX
+        group_name = DDCpt(ad_group(Type,
+                                    (attribute_1_name, attribute_1_suffix),
+                                    (attribute_2_name, attribute_2_suffix),
+                                    ...................
+                                    (attribute_n_name, attribute_n_suffix))
+
+        # ADD ATTRIBUTE PLUGINS AS COMPONENTS USING THE SYNTAX
+        plugin_name = Cpt(PluginClass, suffix = Plugin_suffix+':')
+
+
+.. note::
+    1. DetectorBase can be swapped out for any other Areadetector Device
+    class that inherits from DetectorBase. Also an optional trigger_mixin
+    class can be specifed (trigger_mixin classes must be listed first).
+
+    2. Component (Cpt) can be replaced with ADComponent(ADCpt) if this is a
+    builtin areadetector component as this includes the possibility of using
+    the Areadetector documentation.
+
+    3. PluginClass can be PluginBase, CamBase or any plugin/cam class that
+    inherits from either of these.
+
+
+
+The Areadetector device should then be instantiated using:
+
+.. code-block:: python
+
+    ADdevice_name = Some_Areadetector_Device_Class(Areadetector_device_PV_suffix,
+                                                  name = 'ADdevice_name')
+
+
+Custom Plugins or Cameras
+=========================
+For custom hardware based on area-detector it may be necesary to add a custom
+plugin or camera class, this section will cover what is required. Both
+'plugins' and 'cameras' act in the same way, but have slightly different 'base'
+attributes, hence they have different 'base classes'. New Plugin classes should
+inherit from ``ophyd.areadetector.base.PluginBase`` while new Camera classes
```

Comment:
Write this as :class:`ophyd.areadetector.base.PluginBase` and it will become a link. For more info, see [this section of the cookiecutter docs](https://nsls-ii.github.io/scientific-python-cookiecutter/writing-docs.html#referencing-documented-objects).

Reply:
will update this throughout, thanks for pointing it out.